src/ts/components/seekbarlabel.ts

```diff
@@ -136,31 +136,50 @@ export class SeekBarLabel extends Container<SeekBarLabelConfig> {
       this.thumbnailImageLoader.load(thumbnail.url, (url, width, height) => {
-        let thumbnailCountX = width / thumbnail.w;
-        let thumbnailCountY = height / thumbnail.h;
-
-        let thumbnailIndexX = thumbnail.x / thumbnail.w;
-        let thumbnailIndexY = thumbnail.y / thumbnail.h;
-
-        let sizeX = 100 * thumbnailCountX;
-        let sizeY = 100 * thumbnailCountY;
-
-        let offsetX = 100 * thumbnailIndexX;
-        let offsetY = 100 * thumbnailIndexY;
-
-        let aspectRatio = 1 / thumbnail.w * thumbnail.h;
-
-        // The thumbnail size is set by setting the CSS 'width' and 'padding-bottom' properties. 'padding-bottom' is
-        // used because it is relative to the width and can be used to set the aspect ratio of the thumbnail.
-        // A default value for width is set in the stylesheet and can be overwritten from there or anywhere else.
-        thumbnailElement.css({
-          'display': 'inherit',
-          'background-image': `url(${thumbnail.url})`,
-          'padding-bottom': `${100 * aspectRatio}%`,
-          'background-size': `${sizeX}% ${sizeY}%`,
-          'background-position': `-${offsetX}% -${offsetY}%`,
-        });
+        if (['x', 'y', 'w', 'h'].every((key) => thumbnail.hasOwnProperty(key))) {
+          thumbnailElement.css(this.thumbnailCssSprite(thumbnail, width, height));
+        } else {
+          thumbnailElement.css(this.thumbnailCssSingeImage(thumbnail, width, height));
+        }
       });
     }
   }
+
+  private thumbnailCssSprite(thumbnail: bitmovin.PlayerAPI.Thumbnail, width: number, height: number): any {
+    let thumbnailCountX = width / thumbnail.w;
+    let thumbnailCountY = height / thumbnail.h;
+
+    let thumbnailIndexX = thumbnail.x / thumbnail.w;
+    let thumbnailIndexY = thumbnail.y / thumbnail.h;
+
+    let sizeX = 100 * thumbnailCountX;
+    let sizeY = 100 * thumbnailCountY;
+
+    let offsetX = 100 * thumbnailIndexX;
+    let offsetY = 100 * thumbnailIndexY;
+
+    let aspectRatio = 1 / thumbnail.w * thumbnail.h;
+
+    // The thumbnail size is set by setting the CSS 'width' and 'padding-bottom' properties. 'padding-bottom' is
+    // used because it is relative to the width and can be used to set the aspect ratio of the thumbnail.
+    // A default value for width is set in the stylesheet and can be overwritten from there or anywhere else.
+    return {
+      'display': 'inherit',
+      'background-image': `url(${thumbnail.url})`,
+      'padding-bottom': `${100 * aspectRatio}%`,
+      'background-size': `${sizeX}% ${sizeY}%`,
+      'background-position': `-${offsetX}% -${offsetY}%`,
+    };
+  }
+
+  private thumbnailCssSingeImage(thumbnail: bitmovin.PlayerAPI.Thumbnail, width: number, height: number): any {
+    let aspectRatio = 1 / width * height;
+
+    return {
+      'display': 'inherit',
+      'background-image': `url(${thumbnail.url})`,
+      'padding-bottom': `${100 * aspectRatio}%`,
+      'background-size': `100% 100%`,
```

Comment:
You still need to set `background-position: 0 0` (in case a thumbnail sprite was displayed before)